motionplan/tpspace/ptgDiffDrive.go

```diff
@@ -0,0 +1,74 @@
+package tpspace
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/golang/geo/r3"
+
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+// ptgDiffDrive defines a PTG family composed of circular trajectories with an alpha-dependent radius.
+type ptgDiffDrive struct {
+	maxMMPS float64 // millimeters per second velocity to target
+	maxRPS  float64 // radians per second of rotation when driving at maxMMPS and turning at max turning radius
+}
+
+// NewDiffDrivePTG creates a new PTG of type ptgDiffDrive.
+func NewDiffDrivePTG(maxMMPS, maxRPS float64) PTG {
+	return &ptgDiffDrive{
+		maxMMPS: maxMMPS,
+		maxRPS:  maxRPS,
+	}
+}
+
+// For this particular driver, turns alpha into a linear + angular velocity. Linear is just max * fwd/back.
+// Note that this will NOT work as-is for 0-radius turning. Robots capable of turning in place will need to be special-cased
+// because they will have zero linear velocity through their turns, not max.
+func (ptg *ptgDiffDrive) Velocities(alpha, dist float64) (float64, float64, error) {
```

Comment:
I'm confused how Velocities() will be used. It looks like when dist <= alpha, we just do the spin, and when dist > alpha, we just move straight. Does this mean that each instance of the PTG is either a spin or a move straight? Or does the same PTG get used with different values of alpha and dist?

Reply:
`Velocities` allows us to ask the question, "at this point along the trajectory, what are my instantaneous velocities?".

When we execute the PTG for a given alpha and distance, we start at distance 0 and execute the PTG up to whatever distance. This function is expected to the called many times at some granularity.

Reply:
Thank you!